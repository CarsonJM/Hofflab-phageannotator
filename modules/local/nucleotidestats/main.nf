process NUCLEOTIDESTATS {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/biopython:1.81':
        'biocontainers/biopython:1.81' }"

    input:
    tuple val(meta) , path(fasta)
    tuple val(meta2), path(proteins)

    output:
    tuple val(meta), path("${prefix}_nuc_stats.tsv")    , emit: nuc_stats
    path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    nucleotidestats.py \\
        --input ${fasta} \\
        --proteins ${proteins} \\
        --output ${prefix}_nuc_stats.tsv \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(echo \$(biopython_version.py 2>&1))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_nuc_stats.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(echo \$(biopython_version.py 2>&1))
    END_VERSIONS
    """
}
