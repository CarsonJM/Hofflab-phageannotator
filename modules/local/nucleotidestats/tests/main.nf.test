nextflow_process {

    name "Test Process NUCLEOTIDESTATS"
    script "../main.nf"
    process "NUCLEOTIDESTATS"

    test("fasta.gz + faa.gz") {
        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        file(params.modules_testdata_base_path + "genomics/sarscov2/genome/genome.fasta.gz", checkIfExists: true)
                    ]
                )
                input[1] = Channel.of(
                    [
                        [ id:'test' ],
                        file("${projectDir}/assets/test-datasets/fastas/sarscov2.faa.gz", checkIfExists: true)
                    ]
                )
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }


    test("fasta.gz + faa.gz - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        file(params.modules_testdata_base_path + "genomics/sarscov2/genome/genome.fasta.gz", checkIfExists: true)
                    ]
                )
                input[1] = Channel.of(
                    [
                        [ id:'test' ],
                        file("${projectDir}/assets/test-datasets/fastas/sarscov2.faa.gz", checkIfExists: true)
                    ]
                )
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
