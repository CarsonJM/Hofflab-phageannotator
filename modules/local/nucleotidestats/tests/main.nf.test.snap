{
    "fasta.gz + faa.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_nuc_stats.tsv:md5,3c213a7ff9a188d4d12b66f93ac627cf"
                    ]
                ],
                "1": [
                    "versions.yml:md5,7872a31dc4800bdaa18bc3a0d7ec7bad"
                ],
                "nuc_stats": [
                    [
                        {
                            "id": "test"
                        },
                        "test_nuc_stats.tsv:md5,3c213a7ff9a188d4d12b66f93ac627cf"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7872a31dc4800bdaa18bc3a0d7ec7bad"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-24T13:48:12.705246724"
    },
    "fasta.gz + faa.gz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_nuc_stats.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,7872a31dc4800bdaa18bc3a0d7ec7bad"
                ],
                "nuc_stats": [
                    [
                        {
                            "id": "test"
                        },
                        "test_nuc_stats.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7872a31dc4800bdaa18bc3a0d7ec7bad"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-24T13:48:35.493471274"
    }
}